python/ucxx/_lib/tests/conftest.py

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+
+import pytest
+
+# Prevent calls such as `cudf = pytest.importorskip("cudf")` from initializing
+# a CUDA context. Such calls may cause tests that must initialize the CUDA
+# context on the appropriate device to fail.
+# For example, without `RAPIDS_NO_INITIALIZE=True`, `test_benchmark_cluster`
+# will succeed if running alone, but fails when all tests are run in batch.
+os.environ["RAPIDS_NO_INITIALIZE"] = "True"
+
+
+@pytest.fixture(autouse=True)
+def collect_garbage():
+    import gc
+
+    gc.collect()
```

Comment:
I think the more idiomatic way of doing this is to define a teardown protocol (https://docs.pytest.org/en/7.1.x/_modules/_pytest/hookspec.html#pytest_runtest_teardown)
```
def pytest_runtest_teardown(item, nextitem):
    yield
    gc.collect()
```

(We can move the `import gc` to the top-level).

Reply:
Great suggestion, I forgot about pytest hooks. Done in https://github.com/rapidsai/ucxx/pull/139/commits/bb63249bd25a33632507a1fe1800874ca04c18cb .

Reply:
Sorry, and I made a bork. The `yield` should not be there.